pkg/config/config_parser.go

```diff
@@ -236,3 +241,72 @@ func decodeCloudQueryBlock(block *hcl.Block, ctx *hcl.EvalContext) (CloudQuery,
 	cq.Logger = &logging.GlobalConfig
 	return cq, gohcl.DecodeBody(block.Body, ctx, &cq)
 }
+
+func validateConnection(connection *Connection) diag.Diagnostics {
+	var diags diag.Diagnostics
+	if connection.DSN != "" {
+		// allow using a DSN flag even if the config file has explicitly attributes (user, password, host, database, etc.)
+		dsnFromFlag := viper.GetString("dsn")
+		if dsnFromFlag == "" && connection.IsAnyConnParamsSet() {
+			diags = append(diags, diag.NewBaseError(
+				fmt.Errorf("invalid connection configuration"),
+				diag.USER, diag.WithOptionalSeverity(diag.ERROR),
+				diag.WithDetails("DSN specified along with explicit attributes, only one type is supported")),
+			)
+		}
+	} else {
+		if connection.Host == "" {
+			diags = append(diags, diag.NewBaseError(
+				fmt.Errorf("invalid connection configuration"),
+				diag.USER, diag.WithOptionalSeverity(diag.ERROR),
+				diag.WithDetails("missing host")),
+			)
+		}
+		if connection.Database == "" {
+			diags = append(diags, diag.NewBaseError(
+				fmt.Errorf("invalid connection configuration"),
+				diag.USER, diag.WithOptionalSeverity(diag.ERROR),
+				diag.WithDetails("missing database")),
+			)
+		}
+	}
+
+	return diags
+}
+
+func validateProvidersBlock(config *Config) diag.Diagnostics {
```

Comment:
document this function

Reply:
Done in https://github.com/cloudquery/cloudquery/pull/913/commits/67837b2ad64d2cadb4358963cd78c6414a4fceb7